### DISCUSSION TITLE
Adding semantic highlighting to variables annotated as callables/types

### DISCUSSION THREAD

**Discussion by amotzop** [microsoft/pylance-release#2722] [Enhancement]
The semantic highlighter is really useful for quickly differentiating between different object types (functions, variables, classes, etc.) and to me it especially helps noticing function calls/object creations.
This is why I think it would be nice to add a way to change the semantic highlighting of variables/properties which are annotated as callable (e.g. using the `Callable` annotation, a `Protocol` implementing the `__call__` method, etc.) or annotated as types (i.e., `type[...]`).
This would make it possible to color them differently (especially when they are used as a function/to create a new type instance), and make it easier to see there is a function call happening.
Maybe it will also be useful to add to them an appropriate modifier (so they are colored differently when defined/assigned and when they are called).

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by Hugovdberg** [comment]
I really like this idea. I think this is especially useful when writing code in a functional style where it is common to assign partially applied functions or curried functions to variables. Also, when writing higher order functions, that take callables as inputs, it would be helpful if those callables are highlighted separately from other parameters.
